qiskit/providers/ibmq/experiment/experimentservice.py

```diff
@@ -141,6 +142,10 @@ def experiments(
             group: Filter by hub and group. `hub` must also be specified if `group` is.
             project: Filter by hub, group, and project. `hub` and `group` must also be
                 specified if `project` is.
+            is_public: Filter by public visibility. If True only public experiments
+                will be returned. If False only experiments that are not public will
+                be returned. If None (the default), then a visibility filter is not
+                applied.
```

Comment:
I don't know if using a boolean keyword to represent 3 things is the best approach. Without reading the docs (because who reads docs), a user is likely to think `is_public=True` implies "all experiments visible to me, including public ones". 

Perhaps 2 keywords would make it clearer - `exclude_public` and `public_only`. These are, of course, mutually exclusive so we'll need to check for that. 

In addition, I'd avoid using the word `visibility`, since `share_level` is used previously (to match ibmq jobs), and it's unnecessary to introduce yet another terminology.

Reply:
One more thing - I was going to change the experiment `share_level` to match that of ibmqjob at some point - i.e. `none` instead of `private`, and `global` instead of `public`. I actually like `public` more than `global`, but I like consistency above all :)

Reply:
> Perhaps 2 keywords would make it clearer - `exclude_public` and `public_only`. These are, of course, mutually exclusive so we'll need to check for that.

I [wondered about this myself in the issue](https://github.com/Qiskit/qiskit-ibmq-provider/issues/848#issuecomment-756306601) and while I don't love it, it's fine as long as we validate against using both to create a conflict:

| exclude_public | public_only | result |
| ---------------- | ------------ | ------- |
| False                | False           | no visibility filter (default) |
| True                 | False           | visibility=!public |
| False                | True            | visibility=public |
| True                 | True            | raise error |

Right?

> 
> In addition, I'd avoid using the word `visibility`, since `share_level` is used previously (to match ibmq jobs), and it's unnecessary to introduce yet another terminology.

Ack.